src/otx/algo/object_detection_3d/losses/monodetr_loss.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""main loss for MonoDETR model."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+import torch
+from torch import nn
+from torch.nn import functional
+from torchvision.ops import box_convert
+
+from otx.algo.common.losses.focal_loss import py_sigmoid_focal_loss
+from otx.algo.common.losses.iou_loss import giou_loss
+from otx.algo.object_detection_3d.matchers.matcher_3d import HungarianMatcher3D
+from otx.algo.object_detection_3d.utils.utils import box_cxcylrtb_to_xyxy
+
+from .ddn_loss import DDNLoss
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+
+class MonoDETRCriterion(nn.Module):
+    """This class computes the loss for MonoDETR."""
+
+    def __init__(self, num_classes: int, weight_dict: dict, focal_alpha: float, group_num: int = 11) -> None:
+        """MonoDETRCriterion.
+
+        Args:
+            num_classes: number of object categories, omitting the special no-object category
```

Comment:
```suggestion
            num_classes (int): number of object categories, omitting the special no-object category
```
Could you update type in other parts also?

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/4031/commits/00200f4d4811d82e86e371cb5dabd2541098e83c